### PR TITLE
[feature](pipelineX) support paritition tablet sink shuffle

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: regression-test/suites/nereids_p0/insert_into_table/no_partition.groovy [hunk 1]
```diff
@@ -26,6 +26,7 @@ suite('nereids_insert_no_partition') {
     sql 'set enable_nereids_dml=true'
     sql 'set enable_strict_consistency_dml=true'
 
+    sql 'set experimental_enable_nereids_dml_with_pipeline=false'
     explain {
         // TODO: test turn off pipeline when dml, remove it if pipeline sink is ok
```
Comment (HappenLee) [inline regression-test/suites/nereids_p0/insert_into_table/no_partition.groovy:31]: do remove please

File: be/src/pipeline/exec/exchange_sink_operator.cpp [hunk 2]
```diff
@@ -398,6 +464,40 @@ Status ExchangeSinkOperatorX::sink(RuntimeState* state, vectorized::Block* block
...
+
+            RETURN_IF_ERROR(local_state._row_distribution.generate_rows_distribution(
+                    *block, convert_block, filtered_rows, has_filtered_rows,
+                    local_state._row_part_tablet_ids, local_state._number_input_rows));
+
+            const auto& row_ids = local_state._row_part_tablet_ids[0].row_ids;
+            const auto& tablet_ids = local_state._row_part_tablet_ids[0].tablet_ids;
+            for (int idx = 0; idx < row_ids.size(); ++idx) {
+                const auto& row = row_ids[idx];
+                const auto& tablet_id = tablet_ids[idx];
```
Comment (HappenLee) [inline be/src/pipeline/exec/exchange_sink_operator.cpp:489]: do crc32 hash

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

File: be/src/pipeline/exec/exchange_sink_operator.cpp [hunk 3]
```diff
@@ -398,6 +458,47 @@ Status ExchangeSinkOperatorX::sink(RuntimeState* state, vectorized::Block* block
...
+
+            const auto& row_ids = local_state._row_part_tablet_ids[0].row_ids;
+            const auto& tablet_ids = local_state._row_part_tablet_ids[0].tablet_ids;
+            auto select_rows = row_ids.size();
+            std::vector<uint32_t> crc_hash_vals(select_rows);
+            auto* __restrict crc_hashes = crc_hash_vals.data();
+            for (size_t i = 0; i < select_rows; i++) {
+                crc_hashes[i] =
+                        HashUtil::zlib_crc_hash(&tablet_ids[i], sizeof(int64), crc_hashes[i]);
+            }
```
Comment (HappenLee) [inline be/src/pipeline/exec/exchange_sink_operator.cpp:487]: tablet_ids[i] =
                        HashUtil::zlib_crc_hash(&tablet_ids[i], sizeof(int64), 0);

File: be/src/vec/functions/function_utility.cpp [hunk 4]
```diff
@@ -70,13 +70,23 @@ class FunctionSleep : public IFunction {
...
+    bool use_default_implementation_for_constants() const override { return false; }
+
     Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
                         size_t result, size_t input_rows_count) const override {
         ColumnPtr& argument_column = block.get_by_position(arguments[0]).column;
 
         auto res_column = ColumnUInt8::create();
 
-        if (auto* nullable_column = check_and_get_column<ColumnNullable>(*argument_column)) {
+        if (is_column_const(*argument_column)) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_utility.cpp:81]: why change this code?

File: be/src/vec/sink/vdata_stream_sender.cpp [hunk 5]
```diff
@@ -687,35 +687,43 @@ Status VDataStreamSender::send(RuntimeState* state, Block* block, bool eos) {
...
+                    *block, convert_block, filtered_rows, has_filtered_rows, _row_part_tablet_ids,
+                    _number_input_rows));
+
+            const auto& row_ids = _row_part_tablet_ids[0].row_ids;
+            const auto& tablet_ids = _row_part_tablet_ids[0].tablet_ids;
+            auto select_rows = row_ids.size();
+            std::vector<uint32_t> crc_hash_vals(select_rows);
+            auto* __restrict crc_hashes = crc_hash_vals.data();
+            for (size_t i = 0; i < select_rows; i++) {
+                crc_hashes[i] =
```
Comment (HappenLee) [inline be/src/vec/sink/vdata_stream_sender.cpp:710]: do the same change like upper

File: be/src/vec/functions/function_utility.cpp (hunk 4 above)
Comment (zhangstar333) [reply be/src/vec/functions/function_utility.cpp:81]: have set use_default_implementation_for_constants = false, so need deal with const column

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**